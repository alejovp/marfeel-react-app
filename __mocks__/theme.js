export const themeMock = {
    tipography: {
        primaryFontColor: 'black',
        fontFamily: 'Helvetica, sans-serif'
    },
    header: {
        padding: 15
    },
    sectionMenu: {
        color: 'orange',
        textSize: 1.1 //em
    },
    background: {
        color: 'gray'
    },
    burgerButton: {
        color: 'red'
    }
};
