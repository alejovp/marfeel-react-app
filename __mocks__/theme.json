{
    "tipography": {
        "primaryFontColor": "black",
        "fontFamily": "Helvetica, sans-serif"
    },
    "header": {
        "padding": 15
    },
    "sectionMenu": {
        "color": "orange",
        "textSize": 1.1
    },
    "background": {
        "color": "gray"
    },
    "burgerButton": {
        "color": "red"
    },
    "logo": {
        "image": "https://www.marfeel.com/public/images/landing/header/logo-marfeel.svg",
        "size": 36
    }
}