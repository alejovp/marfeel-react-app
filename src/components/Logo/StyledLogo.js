import styled from 'styled-components';

export const StyledLogo = styled.a`
    text-decoration: none;
    cursor: pointer;
    position: relative;
    img {
        height: ${({ theme }) => theme.logo.size}px;
    }
`;
