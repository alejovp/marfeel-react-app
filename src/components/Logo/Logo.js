import React, { useContext } from 'react';
import { ThemeContext } from 'styled-components';
import { StyledLogo } from './StyledLogo';

export const Logo = () => {
    const { logo } = useContext(ThemeContext);

    return (
        <StyledLogo>
            <img src={logo.image} />
        </StyledLogo>
    );
};
