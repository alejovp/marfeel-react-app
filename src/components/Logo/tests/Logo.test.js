import React from 'react';
import { Logo } from '../Logo';
import theme from '../../../../__mocks__/theme.json';
import { mountWithTheme } from '../../../../scripts/jest/setup';
import { StyledLogo } from '../StyledLogo';


describe('<Logo />', () => {

    const renderedComponent = mountWithTheme(<Logo />, theme);

    describe('shape', () => {

        it('will render correctly', () => {
            expect(renderedComponent).toMatchSnapshot();
        });

        it('will render a <StyledLogo> component', () => {
            expect(renderedComponent.find(StyledLogo)).toHaveLength(1);
        });

        it('will render an <img> elem', () => {
            expect(renderedComponent.find('img')).toHaveLength(1);
        });

    });

    describe('props', () => {

        it('will set the <img> src prop with the theme logo image value', () => {
            expect(renderedComponent.find('img').props().src).toBe(theme.logo.image);
        });

    });

});
