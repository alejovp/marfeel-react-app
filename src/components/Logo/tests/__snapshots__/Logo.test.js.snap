// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> shape will render correctly 1`] = `
<Logo>
  <styled.a>
    <a
      className="kgaXSa"
    >
      <img
        src="https://www.marfeel.com/public/images/landing/header/logo-marfeel.svg"
      />
    </a>
  </styled.a>
</Logo>
`;
